packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -57,10 +57,31 @@ export class OsmosisSwapper implements Swapper<ChainId> {
     this.supportedAssetIds = [cosmosAssetId, osmosisAssetId]
   }
 
-  async getTradeTxs(tradeResult: TradeResult): Promise<TradeTxs> {
-    return {
-      sellTxid: tradeResult.tradeId,
-      buyTxid: tradeResult.tradeId,
+  async getTradeTxs(tradeResult: OsmosisTradeResult): Promise<TradeTxs> {
+    if (tradeResult.cosmosAddress) {
+      const cosmosAdapter = this.deps.adapterManager.get(cosmosChainId) as
+        | cosmos.ChainAdapter
+        | undefined
+
+      if (!cosmosAdapter)
+        throw new SwapError('OsmosisSwapper: couldnt get cosmos adapter', {
+          code: SwapErrorTypes.GET_TRADE_TXS_FAILED,
+        })
+
+      const cosmosTxHistory = await cosmosAdapter.getTxHistory({
+        pubkey: tradeResult.cosmosAddress,
+      })
+      const currentCosmosTxid = cosmosTxHistory?.transactions[0].txid
+
+      return {
+        sellTxid: tradeResult.tradeId,
+        buyTxid: currentCosmosTxid !== tradeResult.previousCosmosTxid ? currentCosmosTxid : '',
```

Comment:
is there any metadata that exists on the tx that can be use to check that it was created as a result of the sell tx? odds are low, but this could point to the incorrect tx if there was a send/receive before buy tx was sent

Reply:
I agree its hacky and it sucks. But I couldnt find a way to do it otherwise and this seemed like the most reasonable fast solution to work in 99% of cases

Reply:
can you drop a comment to state the edge case so we can re-evaluate later if there is a way please